orderer/common/multichannel/registrar.go

```diff
@@ -516,6 +516,45 @@ func (r *Registrar) SwitchFollowerToChain(chainName string) {
 	logger.Infof("Created and started channel %s", cs.ChannelID())
 }
 
+// SwitchChainToFollower creates a follower.Chain from the tip of the ledger and removes the consensus.Chain.
+// It is called when an etcdraft.Chain detects it was evicted from the cluster (i.e. removed from the consenters set)
+// and halts, transferring execution to the follower.Chain.
+func (r *Registrar) SwitchChainToFollower(channelName string) {
+	r.lock.Lock()
+	defer r.lock.Unlock()
+
+	if _, chainExists := r.chains[channelName]; !chainExists {
+		logger.Infof("Channel %s already removed", channelName)
+		return
+	}
+
+	if _, followerExists := r.followers[channelName]; followerExists {
+		logger.Panicf("Programming error, follower already exists: %s", channelName)
+	}
+
+	rl, err := r.ledgerFactory.GetOrCreate(channelName)
+	if err != nil {
+		logger.Panicf("Failed obtaining ledger factory for %s: %v", channelName, err)
+	}
+
+	configBlock := ConfigBlockOrPanic(rl)
+	ledgerRes, clusterConsenter, err := r.initLedgerResourcesClusterConsenter(configBlock)
+	if err != nil {
+		logger.Panicf("Error initializing ledgerResources & clusterConsenter: %s", err)
```

Comment:
The call to `initLedgerResourcesClusterConsenter` will fail if the consenter is not a cluster type, is panic-ing a little heavy handed here?

Reply:
I think this is justified, as `SwitchChainToFollower` is called only from `etcdraft.Chain` hence it is only expected to operate when the consenter is cluster-type. So this condition represents a programming error.